reference/opt/shaders-msl/comp/argument-buffers-image-load-store.msl2.argument.comp

```diff
@@ -0,0 +1,17 @@
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+using namespace metal;
+
+struct spvDescriptorSetBuffer0
+{
+    texture2d<float, access::write> uImage [[id(1)]];
```

Comment:
@billhollings This doesn't work on iOS? ._.

Reply:
Yes...on iOS...we'll have to pull writable textures out via `add_discrete_descriptor_set()`. 

I think we can handle that in the API...especially since the two have to be synchronized. It should be sufficient if SPRIV-Cross detects writability and throw an error.

Reply:
Added a check for this case.